File: SPLTokenCreator/splTokenCreator.js
const anchor = require("@project-serum/anchor");
const anchorSPL = require("@solana/spl-token");
describe("escrowSwap", () => {
  // Configure the client to use the local cluster.
  let provider = anchor.AnchorProvider.env();
  anchor.setProvider(provider);

  it("Is initialized!", async () => {
    // Add your test here.
    const program = anchor.workspace.EscrowSwap;
    let tokenKeyPair = anchor.web3.Keypair.generate();
    await program.rpc.createSplToken({
      accounts: {
        mint: tokenKeyPair.publicKey,
        signer: provider.publicKey,
        rent: anchor.web3.SYSVAR_RENT_PUBKEY,
        tokenProgram: anchorSPL.TOKEN_PROGRAM_ID,
        systemProgram: anchor.web3.SystemProgram.programId,
      },
      signers: [tokenKeyPair],
    });

    let userTokenAccountKeyPair = anchor.web3.Keypair.generate();
    await program.rpc.createUserSplTokenAccount({
      accounts: {
        userSplAccount: userTokenAccountKeyPair.publicKey,
        mint: tokenKeyPair.publicKey,
        signer: provider.publicKey,
        rent: anchor.web3.SYSVAR_RENT_PUBKEY,
        tokenProgram: anchorSPL.TOKEN_PROGRAM_ID,
        systemProgram: anchor.web3.SystemProgram.programId,
      },
      signers: [userTokenAccountKeyPair],
    });

    await program.rpc.mintToken({
      accounts: {
        tokenAccount: tokenKeyPair.publicKey,
        userTokenAccount: userTokenAccountKeyPair.publicKey,
        signer: provider.publicKey,
        tokenProgram: anchorSPL.TOKEN_PROGRAM_ID,
      },
      signers: [],
    });
  });
});

File: SPLTokenCreator/splTokenCreator.rs
use anchor_lang::prelude::*;
use anchor_spl::{
    self,
    token::{Mint, Token, TokenAccount},
};
declare_id!("Fg6PaFpoGXkYsidMpWTK6W2BeZ7FEfcYkg476zPFsLnS");

#[program]
pub mod escrow_swap {
    use anchor_spl::token;

    use super::*;

    pub fn create_spl_token(_ctx: Context<CreateSPLToken>) -> Result<()> {
        Ok(())
    }

    pub fn create_user_spl_token_account(_ctx: Context<CreateUserSPLTokenAccount>) -> Result<()> {
        Ok(())
    }

    pub fn mint_token(ctx: Context<MintToken>, amount: u64) -> Result<()> {
        token::mint_to(
            CpiContext::new(
                ctx.accounts.token_program.to_account_info(),
                token::MintTo {
                    mint: ctx.accounts.token_account.to_account_info(),
                    to: ctx.accounts.user_token_account.to_account_info(),
                    authority: ctx.accounts.signer.to_account_info(),
                },
            ),
            amount,
        )?;
        Ok(())
    }

    pub fn transfer_token(ctx: Context<TransferToken>, amount: u64) -> Result<()> {
        anchor_spl::token::transfer(
            CpiContext::new(
                ctx.accounts.token_program.to_account_info(),
                token::Transfer {
                    from: ctx.accounts.sender_tokens.to_account_info(),
                    to: ctx.accounts.recipient_tokens.to_account_info(),
                    authority: ctx.accounts.sender.to_account_info(),
                },
            ),
            amount,
        )?;

        Ok(())
    }
}

#[derive(Accounts)]
pub struct CreateSPLToken<'info> {
    #[account(init, payer = signer, mint::decimals = 9, mint::authority = signer, mint::freeze_authority = signer)]
    pub mint: Account<'info, Mint>,
    #[account(mut)]
    pub signer: Signer<'info>,
    pub rent: Sysvar<'info, Rent>,
    pub token_program: Program<'info, Token>,
    pub system_program: Program<'info, System>,
}

#[derive(Accounts)]
pub struct CreateUserSPLTokenAccount<'info> {
    #[account(init, payer = signer, token::mint = mint, token::authority = signer)]
    pub user_spl_account: Account<'info, TokenAccount>,
    /// CHECK:` doc comment explaining why no checks through types are necessary.
    #[account(mut)]
    pub mint: AccountInfo<'info>,
    #[account(mut)]
    pub signer: Signer<'info>,
    pub rent: Sysvar<'info, Rent>,
    pub token_program: Program<'info, Token>,
    pub system_program: Program<'info, System>,
}

#[derive(Accounts)]
pub struct MintToken<'info> {
    /// CHECK: This is the token account that we want to mint tokens to
    #[account(mut)]
    pub token_account: UncheckedAccount<'info>,
    /// CHECK: This is the token account that we want to mint tokens to
    #[account(mut)]
    pub user_token_account: UncheckedAccount<'info>,
    #[account(mut)]
    pub signer: Signer<'info>,
    pub token_program: Program<'info, Token>,
}

#[derive(Accounts)]
pub struct TransferToken<'info> {
    pub sender: Signer<'info>,
    #[account(mut)]
    pub sender_tokens: Account<'info, TokenAccount>,
    #[account(mut)]
    pub recipient_tokens: Account<'info, TokenAccount>,
    pub token_program: Program<'info, Token>,
}

/* 
 Note:- We Can Also Implement Other Functions Of Anchor SPL Like Implemented Functions
*/
